packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -138,45 +139,45 @@ export class Snapshot {
     });
   }
 
-  public async waitUntilDone(
-    operationToWaitFor?: ResourceSnapshotsReportType | null,
-    options: WaitUntilDoneOptions = {},
-    iteratee = (_report: ResourceSnapshotsReportModel) => {}
+  public waitUntilDone(
+    operationToWaitFor: ResourceSnapshotsReportType | null,
+    options: Partial<WaitUntilDoneOptions> = {},
+    onRetryCb = (_report: ResourceSnapshotsReportModel) => {}
   ) {
+    const opts = {...Snapshot.defaultWaitOptions, options};
     const toMilliseconds = (seconds: number) => seconds * 1e3;
-    const opts = {
-      ...Snapshot.defaultWaitOptions,
-      ...options,
-    };
-    let timeout: NodeJS.Timeout;
-
-    await new Promise<void>((resolve, reject) => {
-      if (opts.wait > 0) {
-        timeout = setTimeout(() => {
-          const err = new SnapshotOperationTimeoutError(this);
-          clearInterval(interval);
-          reject(err);
-        }, toMilliseconds(opts.wait));
+
+    return retry(
+      this.waitUntilDoneRetryFunction(operationToWaitFor, onRetryCb),
+      {
+        retries: Infinity,
```

Comment:
We don't care about the number of attempt, so `Infinity`

Reply:
Are you sure it supports Infinity? And that you don't have to use `Number.MAX_SAFE_INTEGER`